modules/extension/wmts/src/main/java/org/geotools/ows/wmts/map/WMTSCoverageReader.java

```diff
@@ -734,4 +680,42 @@ private CoordinateReferenceSystem randomSupportedCRS() {
         }
         return result;
     }
+
+    /** Keep on to values for single request */
+    class TileRequest {
+        int width;
+        int height;
+
+        ReferencedEnvelope envelope;
+
+        String time;
+
+        String tileSRS;
+
+        TileRequest(
```

Comment:
So this class accumulates values, which are only used inside the class itself, in order to create a GetTileRequest object almost right after being created... Seems like what a method would do, but in a more complicated way.

If ``initTileRequest`` returned directly a GetTileRequest object, the code would be simpler.

Reply:
I've done it this way, because my intention is to avoid using GetTileRequest. I will go more directly on WMTSTileService.